WireUI/Sources/WireReusableUIComponents/APIExtensions/UIKit/UIBarButtonItem+RightBarButtonItem.swift

```diff
@@ -0,0 +1,72 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import UIKit
+
+public extension UIBarButtonItem {
+
+    /// Creates a right bar button item for a navigation bar with a specified title and action.
+    ///
+    /// This method follows Apple's guidelines for bar button items, using a fixed font size.
+    ///
+    /// - Parameters:
+    ///   - title: The text to display on the button.
+    ///   - action: The action to perform when the button is tapped.
+    ///
+    /// - Returns: A configured `UIBarButtonItem` for use in a navigation bar.
+    ///
+    /// - Note: The font size is fixed at 17 points with a regular weight, adhering to Apple's
+    ///         recommendations for bar button items. This ensures consistency across different
+    ///         device sizes and accessibility settings.
+    ///
+    /// - Note: When the font size is large enough due to accessibility settings, the
+    ///         LargeContentViewer is applied automatically by the system, enhancing
+    ///         readability for users with visual impairments.
+    static func createNavigationRightBarButtonItem(
+        title: String,
+        action: UIAction
+    ) -> UIBarButtonItem {
+        let buttonFont = UIFont.systemFont(ofSize: 17, weight: .regular)
```

Comment:
Is this code any different than that for creating the right side? If not how about extracting the shared code and calling it underneath. If both left and right versions lived in the same file, this extracted code could be private

Reply:
I like the idea. Simplified it quite a lot here: https://github.com/wireapp/wire-ios/commit/d1db8a0e9a8d76a992f851d4766d83b85d591ab1